### ISSUE TITLE
InvalidMacaroonError not very helpful

### ISSUE THREAD

**Issue by henryiii** [pypi/warehouse#10821]
**Describe the bug**

Uploading `awkward` has suddenly started producing this error:

```
Invalid API Token: InvalidMacaroonError('invalid macaroon signature')
```

The really weird thing is that all wheels upload, and this error occurs _only on the SDist_! We've tried regenerating the token, but no change. This also just started happening, with no apparent changes, between RC 3 and 4.

We can try adding `verbose: true` to https://github.com/scikit-hep/awkward-1.0/blob/5e4e4b984a62baf4075ab96113903f49de058ada/.github/workflows/wheels.yml#L144-L146 and see if anything changes.

**Expected behavior**

The SDist should upload too.

**To Reproduce**

https://github.com/scikit-hep/awkward-1.0/actions/workflows/wheels.yml

**My Platform**

GHA & cibuildwheel

**Additional context**

CC @jpivarski

**Comment by di** [comment]
Looking at the build log, it looks like it's failing on the first upload of a wheel for `awkward_cuda_kernels`. Is it possible the API token is only scoped to the `awkward` project?

You can use https://pypi.org/project/pypitoken/ to unpack your token and determine what permissions it has.

**Comment by henryiii** [comment]
Ahh, fantastic, thanks! Didn’t realize this was added in an RC. The error message could be a bit nicer, though, I had no idea what a Macaroon was.

**Comment by di** [comment]
That's a great point, I think we can definitely improve the error message here.

**Comment by ewjoachim** [comment]
(Shameless plug: https://github.com/pypa/warehouse/pull/9264 might solve this problem by changing most of the macaroon-related messages :) )

**Comment by woodruffw** [comment]
We also recently changed some of the plumbing here; it should be possible to make these error messages much nicer in the near future.

Ref: https://github.com/pypa/warehouse/pull/11122#discussion_r849979151

**Comment by woodruffw** [comment]
Fixed by #11885.
